rapid_response_xblock/static/js/src/rapid.js

```diff
@@ -427,21 +448,60 @@
 
       newContainers.merge(containers)
         .attr("class", "chart-container " + (chartKeys.length === 1 ? 'single-chart' : 'two-charts'))
-        .each(function (index, __, charts) {
-          renderChart(d3.select(charts[index]), index);
-        })
         .selectAll("svg")
         .attr("width", calcChartWidth())
         .attr("height", calcChartHeight())
         .selectAll(".chart")
         .attr("transform", "translate(" + ChartSettings.left + "," + ChartSettings.top + ")")
         .select(".xaxis");
 
+      newContainers.each(function (_, index, charts) {
+        renderChart(d3.select(charts[index]), index);
+      });
```

Comment:
Is this loop necessary? What is it doing? It looks like it's just rendering every available chart. Right now each chart is just rendered on demand if I'm not mistaken. Shouldn't that be the only time we need to render a chart?

Reply:
that was redundant code, i missed removing it